base-math/extendddtf.c

```diff
@@ -30,20 +30,63 @@
 #define NAME extend
 #endif
 
+#include <float.h>
+
 #include "dfpacc.h"
 #include "convert.h"
+#include "convert_helpers.h"
 
 CONVERT_WRAPPER(
 // extendddtf, trunctdtf
 	volatile double df_part1, df_part2;
+    int exp;
 	SRC_TYPE dd_part1, dd_part2;
 
+     /* Get normalized exponent.  */
+    (void) getmantandexpd128 (a, &exp, 17, 1e17DL);
+
+    /* Check if the normalized exponent is beyond the exponent table and
+       round correct the result.  */
+    if (exp > BINPOWOF10_LIMIT)
+      {
+	    if (DFP_EXCEPTIONS_ENABLED)
+	      DFP_HANDLE_EXCEPTIONS (FE_OVERFLOW|FE_INEXACT);
+
+        switch (fegetround())
+          {
+            case FE_TOWARDZERO:
+	          return SIGNBIT(a) ? -LDBL_MAX : LDBL_MAX;
+	        case FE_DOWNWARD:
+	          return SIGNBIT(a) ? -INFINITY : LDBL_MAX;
+	        case FE_UPWARD:
+	          return SIGNBIT(a) ? -LDBL_MAX : INFINITY;
+	        case FE_TONEAREST:
+	        default:
+	          return SIGNBIT(a) ? -INFINITY : INFINITY;
+          }
+      }
+    else if (exp < -POWOF10_MIN_DENORM_DBL_EXP) /* Underflow.  */
+      {
+	    if (DFP_EXCEPTIONS_ENABLED)
+	      DFP_HANDLE_EXCEPTIONS (FE_UNDERFLOW|FE_INEXACT);
+
+        switch (fegetround())
+          {
+            case FE_TOWARDZERO:
+	          return SIGNBIT(a) ? -0.0 : 0.0;
+	        case FE_DOWNWARD:
+	          return SIGNBIT(a) ? -__LDBL_DENORM_MIN__ : 0.0;
+	        case FE_UPWARD:
+	          return SIGNBIT(a) ? -0.0 : __LDBL_DENORM_MIN__;
+	        case FE_TONEAREST:
+	        default:
+	          return SIGNBIT(a) ? -0.0 : 0.0;
+          }
+       }
+
 	df_part1 = a;			/* TD -> DF  */
 	dd_part1 = df_part1;		/* DF -> DD./TD.  */
-	dd_part2 = a - dd_part1;	/* DD./TD.  */
-	df_part2 = dd_part2;		/* DD/TD -> DF.  */
```

Comment:
The removal of these lines causes new warning messages:

```
In file included from /home/gabriel/src/libdfp/base-math/extendddtf.c:36:0:
/home/gabriel/src/libdfp/base-math/extendddtf.c: In function ‘__dpd_extendddtf’:
/home/gabriel/src/libdfp/base-math/extendddtf.c:43:21: warning: unused variable ‘dd_part2’ [-Wunused-variable]
  SRC_TYPE dd_part1, dd_part2;
                     ^
/home/gabriel/src/libdfp/include/convert.h:348:4: note: in definition of macro ‘CONVERT_WRAPPER’
    __VA_ARGS__  \
    ^~~~~~~~~~~
/home/gabriel/src/libdfp/base-math/extendddtf.c:43:11: warning: variable ‘dd_part1’ set but not used [-Wunused-but-set-variable]
  SRC_TYPE dd_part1, dd_part2;
           ^
/home/gabriel/src/libdfp/include/convert.h:348:4: note: in definition of macro ‘CONVERT_WRAPPER’
    __VA_ARGS__  \
    ^~~~~~~~~~~
/home/gabriel/src/libdfp/base-math/extendddtf.c:41:28: warning: unused variable ‘df_part2’ [-Wunused-variable]
  volatile double df_part1, df_part2;
                            ^
/home/gabriel/src/libdfp/include/convert.h:348:4: note: in definition of macro ‘CONVERT_WRAPPER’
    __VA_ARGS__  \
    ^~~~~~~~~~~
```

Reply:
@gftg85  When I rebased and force pushed I forgot to add those lines back. Remove those lines was the part of my first commit and should be not on this commit. I've made another push. I guess everything is fine now. No warning and no failing tests on bfp-conversions.